### ISSUE TITLE
Layout grid using EPSG 4326 does not draw when project's CRS is different

### ISSUE THREAD

**Issue by spluque** [qgis/QGIS#50411]
### What is the bug or the crash?

If a project is set up with a chosen CRS that is different from the layers included in it, it is impossible to draw a grid in a layout using another CRS (such as EPSG:4326) and intervals selected with map units. Nothing is drawn, no matter what interval is specified.

### Steps to reproduce the issue

1. Set up a project with a CRS that is not EPSG:4326 (e.g. EPSG:4954);
2. Add any layer with a CRS that differs from the project's (e.g. EPSG:4326) without changing the project's CRS;
3. Create a layout and add a grid using CRS EPSG:4326, with map units intervals at any desired intervals in degrees;
4. Nothing is drawn.

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

QGIS version | 3.22.11-Białowieża | QGIS code branch | Release 3.22
-- | -- | -- | --
Qt version | 5.15.4
Python version | 3.10.7
Compiled against GDAL/OGR | 3.5.1 | Running against GDAL/OGR | 3.5.2
PROJ version | 9.1.0
EPSG Registry database version | v10.074 (2022-08-01)
GEOS version | 3.11.0-CAPI-1.17.0
SQLite version | 3.39.3
PostgreSQL client version | 14.5 (Debian 14.5-2)
SpatiaLite version | 5.0.1
QWT version | 6.1.4
QScintilla2 version | 2.11.6
OS version | Debian GNU/Linux bookworm/sid
  |   |   |  
Active Python plugins
shapetools | 3.4.6
sagaprovider | 2.12.99
MetaSearch | 0.3.5
processing | 2.12.99
grassprovider | 2.12.99
db_manager | 0.1.20

</body></html>QGIS version
3.22.11-Białowieża
QGIS code branch
[Release 3.22](https://github.com/qgis/QGIS/tree/release-3_22)
Qt version
5.15.4
Python version
3.10.7
Compiled against GDAL/OGR
3.5.1
Running against GDAL/OGR
3.5.2
PROJ version
9.1.0
EPSG Registry database version
v10.074 (2022-08-01)
GEOS version
3.11.0-CAPI-1.17.0
SQLite version
3.39.3
PostgreSQL client version
14.5 (Debian 14.5-2)
SpatiaLite version
5.0.1
QWT version
6.1.4
QScintilla2 version
2.11.6
OS version
Debian GNU/Linux bookworm/sid

Active Python plugins
shapetools
3.4.6
sagaprovider
2.12.99
MetaSearch
0.3.5
processing
2.12.99
grassprovider
2.12.99
db_manager
0.1.20

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by roya0045** [comment]
Works fine on windows. Unique to linux?

**Comment by agiudiceandrea** [comment]
@spluque, if you create a new project with "a CRS that is not EPSG:4326 (e.g. EPSG:4954)" and afterward add "a CRS that differs from the project's (e.g. EPSG:4326)" then the project CRS will automatically revert back to EPSG:4326. Is this your case? If yes, then I cannot replicate the reported issue neither on Windows or Ubuntu/Linux. Could you please clarify?

**Comment by spluque** [comment]
@agiudiceandrea yes, I've updated my recipe to indicate resetting to the CRS in step (1) after adding the layer in step (2).

**Comment by agiudiceandrea** [comment]
I can confirm the issue on both Windows and Ubuntu/Linux: using a map project with EPSG:4954 CRS, regardless of the layer CRS in the map, it is impossible to display an EPSG:4326 grid on the map (with the same CRS of the map project) in a layout.
This issue doesn't occur with other map CRSs like e.g. EPSG:3857.

**Comment by phidrho** [comment]
I could get the grid to show on QGIS 3.32.0-Lima, while moving around, sometimes is shown sometimes not. See images:

![image](https://github.com/qgis/QGIS/assets/1849293/554ea421-1413-4938-b9d8-eb2cb4f4ebce)

![image](https://github.com/qgis/QGIS/assets/1849293/391831ae-73cd-4599-ab61-f9fa17515c3e)

**Comment by tombrennan06** [comment]
This looks similar to the issue I have. 

I have a layout in EPSG:32756 (WGS/UTM Zone 56S) and a grid in EPSG:4326.

![image](https://github.com/qgis/QGIS/assets/2381307/8bd72e22-368e-453c-a0d1-c859da0a37dd)

It appears that the grid lines are wrapping across the image, causing additional lines to be created that shouldn't be there. Hard to be certain this is the same root cause as the original issue above.
